test/e2e/reconciler/locationworkspace/rootcompute_test.go

```diff
@@ -153,4 +163,80 @@ func TestRootComputeWorkspace(t *testing.T) {
 		}
 		return true, ""
 	}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+	t.Logf("Wait for being able to list Deployments in the user workspace")
+	require.Eventually(t, func() bool {
+		_, err := kubeClusterClient.Cluster(consumerWorkspace).AppsV1().Deployments("default").List(ctx, metav1.ListOptions{})
+		if errors.IsNotFound(err) {
+			t.Logf("service err %v", err)
+			return false
+		} else if err != nil {
+			t.Logf("service err %v", err)
+			return false
+		}
```

Comment:
Given there is no different behavior for different types of errors, I would recommend collapsing this down to only checking for an error

Reply:
Copy of the above code as well ;-)

Reply:
fixed